### PR TITLE
custom_build should allow 0 deps. Also, clean up path unpacking

### PR DESCRIPTION
**Pull request by nicks**
Hello @landism,

Please review the following commits I made in branch nicks/ch10018:

04f72bc58ba682b3bf7c118425d0ccf50997693a (2020-11-10 17:09:28 -0500)
custom_build should allow 0 deps. Also, clean up path unpacking

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/tiltfile/value/path.go [hunk 1]
```diff
@@ -24,3 +26,53 @@ func (p *LocalPath) Unpack(v starlark.Value) error {
...
+	}
+
+	defer iter.Done()
+
+	values := []string{}
+	var item starlark.Value
+	for iter.Next(&item) {
+		str, err := ValueToAbsPath(p.t, item)
+		if err != nil {
+			return fmt.Errorf("unpacking list item #%d: %v", len(values), err)
```
Comment (landism) [inline internal/tiltfile/value/path.go:72]:
```suggestion
			return fmt.Errorf("unpacking list item at index %d: %v", len(values), err)
```

Comment (landism) [reply internal/tiltfile/value/path.go:72]: maybe just me,  but I think "index" is more unambiguously 0-based - without other information, I'd bias towards interpreting "#%d" as 1-based

And/or putting the value of `item.String()` in the error message would make clear which element is problematic. (though that might be better done inside of `ValueToAbsPath`)

File: internal/tiltfile/value/path.go [hunk 2]
```diff
@@ -24,3 +26,53 @@ func (p *LocalPath) Unpack(v starlark.Value) error {
...
+	}
+
+	var iter starlark.Iterator
+	switch x := v.(type) {
+	case *starlark.List:
+		iter = x.Iterate()
+	case starlark.Tuple:
+		iter = x.Iterate()
+	default:
+		return fmt.Errorf("value should be a string or List or Tuple of strings, but is of type %s", v.Type())
```
Comment (landism) [inline internal/tiltfile/value/path.go:62]: "string or" should only be included if `p.AllowsSingleString`

Alternatively, I'd be in favor of deleting that option and always behaving as if it's true. AFAIK the only reason we are inconsistent on whether a single string is allowed is a lack of attention paid to consistency (and the fact that we previously had this logic duplicated everywhere and no centralized place to standardize the behavior). Maybe I'm missing something or maybe changing the existing API is more scope than you want in this PR, though.

**Review by landism — APPROVED** [review]

File: internal/tiltfile/value/path.go (hunk 1 above)
Comment (nicks) [reply internal/tiltfile/value/path.go:72]: i agree!

File: internal/tiltfile/value/path.go (hunk 2 above)
Comment (nicks) [reply internal/tiltfile/value/path.go:59]: ok, i buy that argument!

**Pull request merged**